x/lazyledgerapp/types/payformessage.go

```diff
@@ -0,0 +1,294 @@
+	NamespaceIDSize                        = core.NamespaceSize
+)
+
+///////////////////////////////////////
+// 	MsgWirePayForMessage
+///////////////////////////////////////
+
+var _ sdk.MsgRequest = &MsgWirePayForMessage{}
+
+func (msg *MsgWirePayForMessage) Route() string { return RouterKey }
+
+func (msg *MsgWirePayForMessage) Type() string { return TypeMsgPayforMessage }
+
+// ValidateBasic checks for valid namespace length, declared message size, share
+// commitments, signatures for those share commitments, and fulfills the sdk.Msg
+// interface
+func (msg *MsgWirePayForMessage) ValidateBasic() error {
+	pubK := msg.PubKey()
+
+	// ensure that the namespace id is of length == 8
+	if len(msg.GetMessageNameSpaceId()) != NamespaceIDSize {
+		return fmt.Errorf(
+			"invalid namespace length: got %d wanted 8",
+			len(msg.GetMessageNameSpaceId()),
+		)
+	}
+
+	// ensure that the included message is evenly divisble into shares
+	if uint64(len(msg.GetMessage()))%ShareSize != 0 {
+		return fmt.Errorf("Share message must be divisible by %d", ShareSize)
+	}
+
+	// make sure that the message size matches the actual size of the message
+	if msg.MessageSize != uint64(len(msg.Message)) {
+		return fmt.Errorf(
+			"Declared Message size does not match actual Message size, %d vs %d",
+			msg.MessageSize,
+			len(msg.Message),
+		)
+	}
+
+	for _, commit := range msg.MessageShareCommitment {
+		// check that each commit is valid
+		calculatedCommit, err := CreateCommit(commit.K, msg.GetMessageNameSpaceId(), msg.Message)
+		if err != nil {
+			return err
+		}
+
+		if string(calculatedCommit) != string(commit.ShareCommitment) {
+			return fmt.Errorf("invalid commit for square size %d", commit.K)
+		}
+
+		// check that the signatures are valid
+		bytesToSign, err := msg.GetCommitSignBytes(commit.K)
+		if err != nil {
+			return err
+		}
+
+		if !pubK.VerifySignature(bytesToSign, commit.Signature) {
+			return fmt.Errorf("invalid signature for share commitment to square size %d", commit.K)
+		}
+	}
+
+	return nil
+}
+
+// GetSignBytes returns messages bytes that need to be signed in order for the
+// message to be valid
+func (msg *MsgWirePayForMessage) GetSignBytes() []byte {
+	out, err := msg.GetCommitSignBytes(SquareSize)
+	if err != nil {
+		// this panic can only be reached if the nmt cannot push bytes onto the
+		// tree while creating the commit. This should never happen, as an error
+		// only occurs when out of order or varying sized namespaces are used,
+		// and we are using an identical namespace when pushing to the nmt
+		// https://github.com/lazyledger/nmt/blob/b22170d6f23796a186c07e87e4ef9856282ffd1a/nmt.go#L250
+		panic(err)
+	}
+	return out
+}
+
+// GetSigners returns the addresses of the message signers
+func (msg *MsgWirePayForMessage) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{sdk.AccAddress(msg.PubKey().Address().Bytes())}
+}
+
+// PubKey uses the string version of the public key to
```

Comment:
The public key to....???

Reply:
Addressed in https://github.com/lazyledger/lazyledger-app/pull/21/commits/07ac37e637166015864f721b4e38071fbbb73611